### ISSUE TITLE
SentryError object has no attribute 'message'

### ISSUE THREAD

**Issue by kwmonroe** [juju/amulet#165]
We started raising SentryErrors on things like `upload_scripts()` in https://github.com/juju/amulet/pull/164

Unfortunately, the exception handler is failing in recent cwr runs, eg:

http://data.vapour.ws/cwr-tests/results/bundle_hadoop_kafka/f8a693ff98bb46f0855356cb37a556ae/report.html

```
                                
                                                                            2016-11-09 00:10:32 Starting deployment of cwr-azure:admin/cwr-azure
2016-11-09 00:10:32 Deploying applications...
2016-11-09 00:11:40 Config specifies num units for subordinate: rsyslog-forwarder-ha
2016-11-09 00:11:41 Adding relations...
2016-11-09 00:11:41 Deployment complete in 69.54 seconds
ERROR exit status 1 (/usr/local/lib/python3.5/dist-packages/amulet/unit-scripts/amulet/__pycache__: not a regular file)
E
======================================================================
ERROR: setUpClass (__main__.RsyslogForwarder)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/amulet/sentry.py", line 84, in fromunitdata
    unitsentry.upload_scripts()
  File "/usr/local/lib/python3.5/dist-packages/amulet/sentry.py", line 126, in upload_scripts
    subprocess.check_call(cmd.split())
  File "/usr/lib/python3.5/subprocess.py", line 581, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['juju', 'scp', '/usr/local/lib/python3.5/dist-packages/amulet/unit-scripts/amulet/__pycache__', 'rsyslog/0:/tmp/amulet']' returned non-zero exit status 1

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/amulet/deployer.py", line 708, in setup
    self.services, timeout=timeout, juju_env=self.juju_env)
  File "/usr/local/lib/python3.5/dist-packages/amulet/sentry.py", line 364, in __init__
    self.unit[unit] = UnitSentry.fromunitdata(unit, unit_data)
  File "/usr/local/lib/python3.5/dist-packages/amulet/sentry.py", line 86, in fromunitdata
    raise SentryError('Unable to upload scripts')
amulet.sentry.SentryError: Unable to upload scripts

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/tmp/cwr-tmp-XysoDd/tmpRxuVQd/rsyslog-forwarder-ha/tests/10_deploy_test.py", line 22, in setUpClass
    cls.d.setup(timeout=600)
  File "/usr/local/lib/python3.5/dist-packages/amulet/deployer.py", line 710, in setup
    raise_status(INFRA_FAIL, msg=e.message)
AttributeError: 'SentryError' object has no attribute 'message'

----------------------------------------------------------------------
Ran 0 tests in 76.280s

FAILED (errors=1)
```

**Comment by jamesbeedy** [comment]
@kwmonroe I was hitting this earlier .... python2 fixes

**Comment by kwmonroe** [comment]
Thanks @jamesbeedy -- i'm not sure if py2 is available in cloud-weather-report.  I *think* the fix here is just to pass `msg=e` instead of `msg=e.message` like this:

```diff
            except SentryError as e:
-                raise_status(INFRA_FAIL, msg=e.message)
+                raise_status(INFRA_FAIL, msg=e)
```

**Comment by jamesbeedy** [comment]
oooh niceee

**Comment by sparkiegeek** [comment]
Can we get a release with this fix in? AFAICS 1.18.0 doesn't have this fix
